transformers/pipelines.py

```diff
@@ -94,32 +95,71 @@ def __call__(self, *texts, **kwargs):
 
 
 class QuestionAnsweringPipeline(Pipeline):
+    """
+    Question Answering pipeling involving Tokenization and Inference.
+    TODO:
+     - top-k answers
+     - return start/end chars
+     - return score
+    """
+
+    def __init__(self, model, tokenizer: Optional[PreTrainedTokenizer]):
+        super().__init__(model, tokenizer)
+
+    @staticmethod
+    def create_sample(question: Union[str, List[str]], context: Union[str, List[str]]) -> Union[dict, List[Dict]]:
+        is_list = isinstance(question, list)
+
+        if is_list:
+            return [{'question': q, 'context': c} for q, c in zip(question, context)]
+        else:
+            return {'question': question, 'context': context}
 
     @classmethod
     def from_config(cls, model, tokenizer: PreTrainedTokenizer, **kwargs):
         pass
 
-    def __call__(self, texts, **kwargs):
+    def __call__(self, *texts, **kwargs):
+        # Set defaults values
+        kwargs.setdefault('max_answer_len', 15)
+        kwargs.setdefault('topk', 1)
+
+        if kwargs['topk'] < 1:
+            raise ValueError('topk parameter should be >= 1 (got {})'.format(kwargs['topk']))
+
+        if kwargs['max_answer_len'] < 1:
+            raise ValueError('max_answer_len parameter should be >= 1 (got {})'.format(kwargs['max_answer_len']))
+
+        # Tabular input
+        if 'question' in kwargs and 'context' in kwargs:

```

Comment:
Nice.
We should mention explicitly the priority order among arguments in the docstring

Reply:
Maybe it would be more natural to have this order?
positional arguments (`*texts`) > `X` > `data` > `question` + `answer`.

Reply:
Done in 9a24e0c